### PR TITLE
Replace Array.GetLength on single-dimensional arrays with Length property

### PR DESCRIPTION
**Pull request by h3xds1nz**
## Description

Replaces `Array.GetLength(0)` on single-dimensional arrays with the `Length` property for performance benefit, clarity and general code quality. I didn't replace a few in `Color` and `ColorTransform` as that's part of my other PR and I don't want to create conflicts.

## Customer Impact

Improved performance.

## Regression

No.

## Testing

Build, the changes are minimal hence no further testing.

## Risk

Low.

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/wpf/pull/9334)

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Color.cs
Comment (hiteshwpfmsft) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Color.cs:1]: Did we miss line number 1113?
(this.nativeColorValue.GetLength(0) > 0)

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs [hunk 1]
```diff
@@ -1085,7 +1085,7 @@ internal override void FinalizeCreation()
 
             if (sourceBuffer.Rank == 1)
             {
-                if (sourceBuffer.GetLength(0) <= 0)
+                if (sourceBuffer.Length == 0)
```
Comment (hiteshwpfmsft) [inline src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs:1088]: For one dimensional arrays I would prefer to use "Length"
but would like to encourage "GetLength" for multidimensional arrays for consistency reasons.

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Color.cs
Comment (h3xds1nz) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Color.cs:1]: It is part of #9330, I didn't wanna create conflicts.

File: src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs (hunk 1 above)
Comment (h3xds1nz) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs:1088]: We can do it for consistency reasons, I'll push it later today. I'll cache the result of `GetLength` as it is queried twice though.

Comment (h3xds1nz) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs]: @hiteshwpfmsft Thanks, I've also changed the `<=` compares to `==` there as it makes no sense to compare less than while at it.

Comment (hiteshwpfmsft) [reply src/Microsoft.DotNet.Wpf/src/PresentationCore/System/Windows/Media/Imaging/WriteableBitmap.cs]: Agree!

**Review by hiteshwpfmsft — APPROVED** [review]

**Review by dipeshmsft — APPROVED** [review]

**Comment by harshit7962** [comment]
@h3xds1nz Thank you for your contributions.

**Comment by h3xds1nz** [comment]
@harshit7962 Thanks for including them.

**Pull request merged**